java/core/src/main/java/co/worklytics/psoxy/storage/StorageHandler.java

```diff
@@ -0,0 +1,56 @@
+package co.worklytics.psoxy.storage;
+
+import co.worklytics.psoxy.*;
+import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.StorageEventRequest;
+import co.worklytics.psoxy.gateway.StorageEventResponse;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+
+import javax.inject.Inject;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Date;
+
+@NoArgsConstructor(onConstructor_ = @Inject)
+public class StorageHandler {
+
+    private static final DateFormat FORMATTER = new SimpleDateFormat("yyyy/MM/dd");
+
+    @Inject
+    ConfigService config;
+    @Inject
+    Rules rules;
+    @Inject
+    FileHandlerStrategy fileHandlerStrategy;
+    @Inject
+    SanitizerFactory sanitizerFactory;
+
+    private volatile Sanitizer sanitizer;
+    private final Object $writeLock = new Object[0];
+
+    private Sanitizer loadSanitizerRules() {
+        if (this.sanitizer == null) {
+            synchronized ($writeLock) {
+                if (this.sanitizer == null) {
+                    this.sanitizer = sanitizerFactory.create(sanitizerFactory.buildOptions(config, rules));
+                }
+            }
+        }
+        return this.sanitizer;
+    }
+
+    @SneakyThrows
+    public StorageEventResponse handle(StorageEventRequest request) {
+
+        FileHandler fileHandler = fileHandlerStrategy.get(request.getSourceBucketPath());
+        this.sanitizer = loadSanitizerRules();
+
+        return StorageEventResponse.builder()
+                .destinationBucketName(request.getDestinationBucket())
+                .bytes(fileHandler.handle(request.getReaderStream(), sanitizer))
+                .destinationPath(String.format("%s/%s", FORMATTER.format(Date.from(Instant.now())), request.getSourceBucketPath()))
```

Comment:
not sure if these paths are later the most useful to deal with `yyyy/mm/dd/file.csv` but ok, can change later.
problem here is when reading from Worklytics we need to recurse rather than just read all files in one folder